app/storage/storage.go

```diff
@@ -1,11 +1,130 @@
+// Package storage provides a storage engine for sql databases.
+// The storage engine is a wrapper around sqlx.DB with additional functionality to work with the various types of database engines.
+// Each table is represented by a struct, and each struct has a method to work the table with business logic for this data type.
 package storage
 
 import (
+	"context"
+	"io"
+	"sync"
+	"time"
+
 	"github.com/jmoiron/sqlx"
 	_ "modernc.org/sqlite" // sqlite driver loaded here
 )
 
+// EngineType is a type of database engine
+type EngineType string
+
+// enum of supported database engines
+const (
+	EngineTypeUnknown EngineType = ""
+	EngineTypeSqlite  EngineType = "sqlite"
+)
+
+// Engine is a wrapper for sqlx.DB with type.
+// EngineType allows distinguishing between different database engines.
+type Engine struct {
+	sqlx.DB
+	gid    string     // group id, to allow per-group storage in the same database
+	dbType EngineType // type of the database engine
+}
+
 // NewSqliteDB creates a new sqlite database
-func NewSqliteDB(file string) (*sqlx.DB, error) {
-	return sqlx.Connect("sqlite", file)
+func NewSqliteDB(file, gid string) (*Engine, error) {
+	db, err := sqlx.Connect("sqlite", file)
+	if err != nil {
+		return &Engine{}, err
+	}
+	if err := setSqlitePragma(db); err != nil {
+		return &Engine{}, err
+	}
+	return &Engine{DB: *db, gid: gid, dbType: EngineTypeSqlite}, nil
+}
+
+// GID returns the group id
+func (e *Engine) GID() string {
+	return e.gid
+}
+
+// Type returns the database engine type
+func (e *Engine) Type() EngineType {
+	return e.dbType
+}
+
+// MakeLock creates a new lock for the database engine
+func (e *Engine) MakeLock() RWLocker {
+	if e.dbType == EngineTypeSqlite {
+		return &sync.RWMutex{} // sqlite need locking
+	}
+	return &NoopLocker{} // other engines don't need locking
+}
+
+func setSqlitePragma(db *sqlx.DB) error {
+	// set pragmas for sqlite
+	pragmas := map[string]string{
+		// "journal_mode": "WAL",
```

Comment:
This commented code looks weird, I propose to either add comment "example:" and URL with parameters as a reference, or remove it altogether.